### PR TITLE
Jorge/asp candidate generation2

### PR DESCRIPTION
**Pull request by jorgefandinno**
Candidates are now creates in the asp program.

Please check and merge if you find no flaws.

### CODE REVIEW

File: src/eclingo/solver/generator.py [hunk 1]
```diff
@@ -55,30 +60,41 @@ def __call__(self) -> Iterator[Candidate]:
                 fact(SA).
 
             hold(KA) :- kp_hold(KA).
+
+            positive_extra_assumptions(A) :- epistemic_atom(k(A), KA), kp_hold(KA).
+            % negative_extra_assumptions(A) :- epistemic_atom(k(A), KA), kp_not_hold(KA).
```
Comment (lute47lillo) [inline src/eclingo/solver/generator.py:65]: Is this left out in purpose? If so, why still call the #show negative_extra_assumptions(A)?

File: src/eclingo/solver/generator.py [hunk 2]
```diff
@@ -55,30 +60,41 @@ def __call__(self) -> Iterator[Candidate]:
...
+
+            #show positive_extra_assumptions/1.
+            #show negative_extra_assumptions/1.
             """
 
-        self.control.add("base", [], self.reified_program)
```
Comment (lute47lillo) [inline src/eclingo/solver/generator.py:60]: If changed to just reified_program, we can remove the assignment on line 18.

File: src/eclingo/solver/tester.py [hunk 3]
```diff
@@ -57,18 +57,26 @@ def __call__(self, candidate: Candidate) -> bool:
...
 
-        for literal in candidate[0]:
+        for literal in candidate.pos:
             assumption = (literal, True)
             candidate_assumptions.append(assumption)
             literal = literal.arguments[0]
             candidate_pos.append(literal)
 
-        for literal in candidate[1]:
+        for literal in candidate.extra_assumptions.pos:
```
Comment (lute47lillo) [inline src/eclingo/solver/tester.py:66]: What is the reason of creating extra_assumptions? That is not something that was done before, right?

**Review by lute47lillo — COMMENTED** [review]
I have reviewed it and have found no fundamental problems. It works as expected. But I have left some NIT comments as well as a clarification on why using extra_assumptions.

File: src/eclingo/solver/generator.py (hunk 2 above)
Comment (jorgefandinno) [reply src/eclingo/solver/generator.py:60]: We can do that

File: src/eclingo/solver/tester.py (hunk 3 above)
Comment (jorgefandinno) [reply src/eclingo/solver/tester.py:66]: I was thinking on a future optimization and it slip there. It does not bother, but it actually does nothing so far. Just let it there, for now. We need to star benchmarking before moving to that optimization. The way to test and optimization is by benchmarking and see that action improves time, as the last did with yale shooting.

**Review by lute47lillo — APPROVED** [review]

**Pull request merged**